pkg/transforms/http_test.go

```diff
@@ -113,3 +177,23 @@ func TestHTTPPostInvalidParameter(t *testing.T) {
 	assert.Equal(t, "marshaling input data to JSON failed, "+
 		"passed in data must be of type []byte, string, or support marshaling to JSON", result.(error).Error())
 }
+
+type mockSecretClient struct {
+}
+
+// NewMockSecretProvider provides a mocked version of the mockSecretClient to avoiding using vault in our tests
+func newMockSecretProvider(loggingClient logger.LoggingClient, configuration *common.ConfigurationStruct) *security.SecretProvider {
+	mockSP := security.NewSecretProvider(logClient, config)
+	mockSP.SecretClient = &mockSecretClient{}
+	return mockSP
+}
+
+// GetSecrets mock implementation of GetSecrets
+func (s *mockSecretClient) GetSecrets(path string, keys ...string) (map[string]string, error) {
+	return map[string]string{"Secret-Header-Name": "value"}, nil
```

Comment:
Add returning error if path or key not expected so can test secret not in vault.